plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/StandardDeployPreferencesPanel.java

```diff
@@ -465,4 +452,29 @@ public void setFont(Font font) {
     expandableComposite.setFont(font);
     FontUtil.convertFontToBold(expandableComposite);
   }
+
+  private class AccountSelectedChecker implements IValidator {
+    @Override
+    public IStatus validate(Object value /* email */) {
+      if (requireValues && Strings.isNullOrEmpty((String) value)) {
+        if (accountSelector.isSignedIn()) {
+          return ValidationStatus.error(Messages.getString("error.account.missing.signedin"));
+        } else {
+          return ValidationStatus.error(Messages.getString("error.account.missing.signedout"));
+        }
+      }
+      return ValidationStatus.ok();
+    }
+  };
+
+  private class ModelToAccountSelectorValidator implements IValidator {
+    @Override
+    public IStatus validate(Object value /* email */) {
+      // Return OK, because if only one account is signed in, accountSelector will auto-select it.
+      if (accountSelector.getAccountCount() == 1) {
+        return ValidationStatus.ok();
+      }
+      return new AccountSelectedChecker().validate(value);
```

Comment:
For simplicity, if I have to answer. I'll make it accept the instance through the constructor.